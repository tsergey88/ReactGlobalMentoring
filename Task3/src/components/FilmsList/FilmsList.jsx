import React from "react";
import { FilmPreview } from "../FilmPreview";

export class FilmsList extends React.Component {
  state = {
    data: []
  };

  componentDidMount() {
    this.initData();
  }

  initData = async () => {
    let response = await fetch(
      "https://reactjs-cdp.herokuapp.com/movies?searchBy=title&limit=18"
    );
    response = await response.json();
    const data = response.data;
    this.setState({ data });
  };

  render() {
    const { data } = this.state;
    return (
      <div className="container">
        {data.map(item => (
          <FilmPreview key={item.id} {...item} />
        ))}
      </div>
    );
  }
}
