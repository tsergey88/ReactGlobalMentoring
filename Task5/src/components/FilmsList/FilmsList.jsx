import React from "react";
import { connect } from "react-redux";

import { fetchFilms } from "../../actions/FilmsAction";
import { FilmPreview } from "../FilmPreview";

class FilmsList extends React.Component {
  componentDidMount() {
    this.props.fetchFilms(18);
  }

  render() {
    const { data, isLoading } = this.props;
    if (isLoading) {
      return <div className="container">Loading...</div>;
    }
    return (
      <div className="container">
        {data.map(item => (
          <FilmPreview key={item.id} {...item} />
        ))}
      </div>
    );
  }
}

const mapStateToProps = state => ({ ...state.FilmsReducer });

const mapDispatchToProps = {
  fetchFilms
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilmsList);
