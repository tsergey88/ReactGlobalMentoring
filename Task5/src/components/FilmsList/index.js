export FilmsList from "./FilmsList.jsx";
