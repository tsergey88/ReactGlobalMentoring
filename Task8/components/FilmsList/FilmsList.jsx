// @flow
import React from "react";

import { FilmPreview } from "../FilmPreview";
import { StateLine } from "../StateLine";
import { SearchRadioButton } from "../SearchForm/SearchRadioButton";

import styles from "../../main.scss";

type Props = {
  fetchFilms?: () => {},
  changeSortBy?: () => {},
  changeSearchInputValue?: () => {},
  isLoading?: boolean,
  request?: {},
  limit?: number,
  sortBy?: string,
  searchBy?: string,
  searchValue?: string,
  genres?: string,
  isFilmPage?: boolean,
  match?: {}
};

class FilmsList extends React.Component<Props> {
  handleChange = checkedValue => {
    this.props.changeSortBy(checkedValue);
  };

  renderPanelSortBy = () => (
    <>
      <div>Sort by</div>
      <div>
        <SearchRadioButton
          title="release date"
          id="1"
          name="sortBy"
          isChecked={this.props.sortBy === "release_date"}
          value="release_date"
          onChange={() => this.handleChange("release_date")}
        />
      </div>
      <div>
        <SearchRadioButton
          title="rating"
          id="2"
          name="sortBy"
          isChecked={this.props.sortBy === "vote_average"}
          value="vote_average"
          onChange={() => this.handleChange("vote_average")}
        />
      </div>
    </>
  );

  render() {
    const { request, isLoading, isFilmPage, genres } = this.props;
    if (isLoading) {
      return <div className={styles.content}>Loading...</div>;
    }
    if (!isLoading && !request.data.length) {
      return (
        <div className={styles.content}>
          <h2>No Films Found</h2>
        </div>
      );
    }
    return (
      <>
        <StateLine
          left={
            !isFilmPage
              ? `${request.total} movies found`
              : `Films by genre: ${genres}`
          }
          right={!isFilmPage && this.renderPanelSortBy()}
        />
        <div className={styles.content}>
          {request.data.map(item => (
            <FilmPreview key={item.id} {...item} />
          ))}
        </div>
      </>
    );
  }
}

export default FilmsList;
