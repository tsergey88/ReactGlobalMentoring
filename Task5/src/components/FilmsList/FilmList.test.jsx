import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import { FilmsList } from "./FilmsList";
import DATA from "./__mocks__/data.json";

const initialState = {
  request: DATA,
  isLoading: false,
  sortBy: "vote_average",
  searchBy: "title",
  searchValue: "",
  limit: 18
};
const props = {
  fetchFilms: jest.fn(),
  changeSortBy: jest.fn()
};
const mockStore = configureStore();
let component;
let store;

describe("FilmsList component", () => {
  beforeEach(() => {
    store = mockStore(initialState);
    component = mount(
      <Provider store={store}>
        <FilmsList {...initialState} {...props} />
      </Provider>
    );
  });

  it("check Prop matches with initialState", () => {
    expect(component.find(FilmsList).prop("request")).toEqual(
      initialState.request
    );
  });

  it("FilmsList call componentDidMount", () => {
    expect(component.find(".item-preview")).toHaveLength(18);
  });

  it("FilmsList loading", () => {
    const component = mount(<FilmsList isLoading={true} {...props} />);
    expect(component.find(".content")).toHaveLength(1);
    expect(component.contains(<div className="content">Loading...</div>)).toBe(
      true
    );
  });

  it("FilmsList no films", () => {
    const component = mount(
      <FilmsList isLoading={false} request={{ data: [] }} {...props} />
    );
    expect(component.find(".content")).toHaveLength(1);
    expect(
      component.contains(
        <div className="content">
          <h2>No Films Found</h2>
        </div>
      )
    ).toBe(true);
  });
});
