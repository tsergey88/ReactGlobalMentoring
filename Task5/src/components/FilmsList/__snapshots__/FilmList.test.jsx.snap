// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList component should be render correctly 1`] = `
<div
  className="container"
/>
`;
